### PR TITLE
Use steepest descent without thermostat in tutorial 02

### LINKED ISSUE #3030: Charged system tutorials run energy minimization with thermostat turned on
This is not best practice and probably should not be shown in beginners tutorials. Also in this case it just makes convergence slower.

### PR DESCRIPTION
**Pull request by jngrad**
Fixes #3030

Description of changes:
- cleanup and document the steepest descent algorithm
- reflect recent changes to the Python interface in the tutorials: `mindist()` renamed to `min_dist()`, periodicity takes a list of booleans
- in tutorial 02 charged system:
    - set Langevin thermostat after energy minimization
    - replace adaptive force-cap minimization strategy with steepest descent

### CODE REVIEW

**Comment by jngrad** [comment]
Notes:
- part 1 of the tutorial still stops minimization based on the minimal distance criterion, we can remove that and have a pure steepest descent instead, just like part 2 of the tutorial
- some NaCl scripts use steepest descent with thermostat ([`nacl_units_confined.py`](https://github.com/espressomd/espresso/blob/python/doc/tutorials/02-charged_system/scripts/nacl_units_confined.py), [`nacl_units_confined_vis.py`](https://github.com/espressomd/espresso/blob/python/doc/tutorials/02-charged_system/scripts/nacl_units_confined_vis.py)), or no minimization at all ([`nacl_units.py`](https://github.com/espressomd/espresso/blob/python/doc/tutorials/02-charged_system/scripts/nacl_units.py), [`nacl_units_vis.py`](https://github.com/espressomd/espresso/blob/python/doc/tutorials/02-charged_system/scripts/nacl_units_vis.py))

File: src/core/minimize_energy.cpp [hunk 1]
```diff
@@ -81,11 +69,11 @@ bool steepest_descent_step(const ParticleRange &particles) {
           f += Utils::sqr(p.f.f[j]);
 
           // Positional increment
-          dp = params->gamma * p.f.f[j];
-          if (fabs(dp) > params->max_displacement)
+          auto dp = params->gamma * p.f.f[j];
```
Comment (fweik) [inline src/core/minimize_energy.cpp]: You can use `clamp` to allow for it being const:
```c++
auto const dp = boost::clamp(gamma * f, -max_displacement, max_displacement);
```

**Review by fweik — APPROVED** [review]

File: src/core/minimize_energy.cpp [hunk 2]
```diff
@@ -40,32 +40,20 @@
...
-  int max_steps;
-  double max_displacement;
-};
-
+/** Currently active steepest descent instance */
 static MinimizeEnergyParameters *params = nullptr;
 
-/* Signum of val */
+/** Sign of the argument */
 template <typename T> int sgn(T val) { return (T(0) < val) - (val < T(0)); }
```
Comment (KaiSzuttor) [inline src/core/minimize_energy.cpp]: ```src/utils/include/utils/math/sgn.hpp```

**Comment by jngrad** [comment]
The CI pipeline apparently canceled itself because the merging happened before the staging: [943f3e2c/staging](https://gitlab.icp.uni-stuttgart.de/espressomd/espresso/commit/943f3e2cc28f4c866579729728cae11449932a2b) then [73b71c12/staging-tmp](https://gitlab.icp.uni-stuttgart.de/espressomd/espresso/commit/73b71c12f5612d6c03304ad3929989aafa41992b).

**Comment by jngrad** [comment]
bors retry

**Comment by jngrad** [comment]
bors retry

**Comment by jngrad** [comment]
bors r=fweik

**Comment by KaiSzuttor** [comment]
bros

**Comment by KaiSzuttor** [comment]
bors retry

**Pull request merged**